components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonsManagerAdapterTest.kt

```diff
@@ -98,6 +98,18 @@ class AddonsManagerAdapterTest {
             R.string.mozac_feature_addons_unavailable_section,
             (itemsWithSections[6] as NotYetSupportedSection).title
         )
+
+        val excludedAddons = listOf(recommendedAddon.id)
+        val itemsWithSections2 = adapter.createListWithSections(addons, excludedAddons)
+        assertEquals(6, itemsWithSections2.size)
+        assertEquals(
+            R.string.mozac_feature_addons_recommended_section,
+            (itemsWithSections2[4] as Section).title
+        )
+        assertEquals(
+            R.string.mozac_feature_addons_unavailable_section,
+            (itemsWithSections2[5] as NotYetSupportedSection).title
+        )
```

Comment:
I'm bit confused shouldn't we be checking that the excluded add-on (`recommendedAddon.id`) is not under the recommended section? But we are just checking for the section position. Could you help me to clarify?

Reply:
My logic is that since there was only one recommendedAddon in this test originally, if that was excluded then the recommended section should be null, so that the section size shrinks from 7 to 6, and after the title of "recommended" there is no actual section of recommended addons but the title of next section (NotYetSupported).

Reply:
Got it, I think it's a bit difficult to follow as we are testing multiple things in one test, we could make the intention more clear one of the following:

1) Writing a new tests that only test this new behavior  and which name makes the intention clear.
2) Add a comments to the actual test to clarify the intention and the expected results.

Reply:
I chose to add some comments to avoid to declare all the same variables again.

Reply:
👍🏽